programs/zstdcli.c

```diff
@@ -1213,6 +1227,23 @@ int main(int const argCount, const char* argv[])
         if (adaptMin > cLevel) cLevel = adaptMin;
         if (adaptMax < cLevel) cLevel = adaptMax;
 
+        if (showDefaultCParams) {
+            size_t fileNb;
+            for (fileNb = 0; fileNb < (size_t)filenames->tableSize; fileNb++) {
+                const size_t fileSize = UTIL_isRegularFile(filenames->fileNames[fileNb]) ? (size_t)UTIL_getFileSize(filenames->fileNames[fileNb]) : 0;
```

Comment:
`UTIL_getFileSize()` returns `UTIL_FILESIZE_UNKNOWN` when the file size is unknown. Instead of checking if it is a regular file, check if the file size is `UTIL_FILESIZE_UNKNOWN`.